packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1269,6 +1270,16 @@ function commitSuspenseComponent(finishedWork: Fiber) {
   if (supportsMutation && primaryChildParent !== null) {
     hideOrUnhideAllChildren(primaryChildParent, newDidTimeout);
   }
+
+  if (enableSuspenseCallback) {
+    const suspenseCallback = finishedWork.memoizedProps.suspenseCallback;
+    if (typeof suspenseCallback === 'function') {
+      const thenables: Set<Thenable> | null = (finishedWork.updateQueue: any);
+      suspenseCallback(new Set(thenables));
```

Comment:
I would only invoke this callback if `newState` is non-null. Meaning only when it is in its "fallback state". I think that should be sufficient to reach parity between the persistent and mutable renderers.

Reply:
Yes, it would be fine not to fire the callback when not in the fallback state. The user code will probably just call `then` on the promise to track exactly when they complete and possibly watch the mount/unmount of the child/fallback.

Now do we want to specify, in the test, if the callback should fire if the thenableSet is identical? This means diffing the current/WIP thneableSet. If so should that would need to happen in Commit work before we call. If we don't specify it then when wont know for sure how many calls to expect in more tricky case. For the user code it's not a big deal because the client code will want to track promises individually anyways.

Reply:
I don’t think we should necessarily dedupe because React shouldn’t be required to keep the cache of promises forever so we don’t want to keep that guarantee.

That said, the retryCache check is probably too late atm and we should move to the ReactFiberThrow as a separate optimization which has the side effect of this set not getting duplicates... mostly.